client/jetpack-connect/controller.js

```diff
@@ -157,6 +158,11 @@ export function connect( context, next ) {
 
 	debug( 'entered connect flow with params %o', params );
 
+	if ( retrieveMobileRedirect() && ! userModule.get() ) {
+		// Force login for mobile app flow. App will intercept and prompt native login.
+		return page.redirect( login( { isNative: true, redirectTo: context.path } ) );
+	}
```

Comment:
I would have preferred to add this to the logged-out routes in `index.js`, but the amount of `/connect` routes would make that unwieldy I think :(

Reply:
Just noting that all of the routes using this handler are preceeded by the mobile cookie persist middleware:

https://github.com/Automattic/wp-calypso/blob/aa0cc400827563b90510d8d824c42cf1da9b6c3a/client/jetpack-connect/index.js#L22-L24

https://github.com/Automattic/wp-calypso/blob/aa0cc400827563b90510d8d824c42cf1da9b6c3a/client/jetpack-connect/index.js#L32-L34

https://github.com/Automattic/wp-calypso/blob/aa0cc400827563b90510d8d824c42cf1da9b6c3a/client/jetpack-connect/index.js#L41-L43

https://github.com/Automattic/wp-calypso/blob/aa0cc400827563b90510d8d824c42cf1da9b6c3a/client/jetpack-connect/index.js#L118-L120